src/fs_mem.ts

```diff
@@ -476,13 +476,26 @@ export class File extends Inode {
   readonly: boolean;
 
   constructor(
-    data: ArrayBuffer | SharedArrayBuffer | Uint8Array | Array<number>,
+    data: ArrayBufferLike | ArrayLike<number>,
     options?: Partial<{
       readonly: boolean;
     }>,
   ) {
     super();
-    this.data = new Uint8Array(data);
+    this.data = (() => {
+      if (data instanceof ArrayBuffer) {
+        return new Uint8Array(data);
+      }
+
+      if (
+        typeof SharedArrayBuffer !== "undefined" &&
+        data instanceof SharedArrayBuffer
+      ) {
+        return new Uint8Array(new Uint8Array(data));
```

Comment:
Why is this doing the `new Uint8Array` twice?

Reply:
We shouldn't copy the memory in the `SharedArrayBuffer` case. https://github.com/bjorn3/browser_wasi_shim/pull/35 specifically added support for it to allow sharing the backing memory of a read-only file across workers.

Reply:
That's fair. So in case of non-readonly file backed by `SharedArrayBuffer` initially, it'll still get copied into a resizable `ArrayBuffer` upon first resize in the follow-up patch, if that looks reasonable to you.

Reply:
Yeah, makes sense.